backend/app/usecase/search/search.go

```diff
@@ -66,13 +69,112 @@ func (s Search) searchResource(resource Resource, orderBy order.Order, query Que
 }
 
 func (s Search) searchShortLink(query Query, orderBy order.Order, filter Filter) (Result, error) {
-	return Result{}, nil
+	if query.User == nil {
+		return Result{}, errors.New("user not provided")
+	}
+	if len(query.Query) == 0 {
+		return Result{}, errors.New("query not provided")
+	}
+
+	shortLinks, err := s.getShortLinkByUser(*query.User)
+	if err != nil {
+		return Result{}, err
+	}
+
+	matchedShortLinks := matchShortLinks(shortLinks, query, orderBy)
+
+	filteredShortLinks := filterShortLinks(matchedShortLinks, filter)
+
+	return Result{
+		shortLinks: filteredShortLinks,
+		users:      nil,
+	}, nil
 }
 
 func (s Search) searchUser(query Query, orderBy order.Order, filter Filter) (Result, error) {
 	return Result{}, nil
 }
 
+func (s Search) getShortLinkByUser(user entity.User) ([]entity.ShortLink, error) {
+	aliases, err := s.userShortLinkRepo.FindAliasesByUser(user)
+	if err != nil {
+		return []entity.ShortLink{}, err
+	}
+
+	return s.shortLinkRepo.GetShortLinksByAliases(aliases)
+}
+
+func matchShortLinks(shortLinks []entity.ShortLink, query Query, orderBy order.Order) []entity.ShortLink {
+	var matchedAliasByAnd, matchedAliasByOr, matchedLongLinkByAnd, matchedLongLinkByOr []entity.ShortLink
+	keywords := strings.Split(query.Query, " ")
```

Comment:
To avoid having different levels of abstraction in a single function, it would be good to move this to it's own function.
```suggestion
	keywords := getKeywordsFromQuery(query.Query)
```
Underneath, the function can just return `strings.Split(query.Query, " ")` but the calling function will not need to know about the implementation details. Not a blocker just a suggestion :smiley:

Reply:
I agree!